### ISSUE TITLE
Add ability to add resources from the front-end 

### ISSUE THREAD

**Issue by krosemichael** [ManifoldScholar/manifold#2873]
## Feature Description

When annotating a text with resources for teaching, I would like to be able to add resources from the place in the front end where I want the resource to appear. 

## User Stories

When I'm adding resources to a text that I'm teaching, I find myself opening two tabs - one with the resources page of the back end open and one with the text. I find a place in the text where I think a resource would be helpful for my students, locate it on the web or on my computer,  add it on the resources page tab, and then place it in the text on the text tab. This works, but is cumbersome, and I want a way to add a resource starting from the place in the text where I want the resource to appear. 

## Design Notes

Maybe an "add resource" button on the resource annotation shelf in the front end that takes you to the resource interface in the back end, though I imagine getting back to the place in the text from there is probably the tricky part.

**Comment by mkgold** [comment]
I strongly concur with this ticket. It has come up for me as a cumbersome process repeatedly as I've been demoing Manifold. It would be great to be able to add a resource from the front end interface or to make some easier connection between front end and back end, as Krysia notes in the design notes

**Comment by zdavis** [comment]
For this to work, the creation of the resource would probably need to happen in a drawer within the reader. I don't think this would be a particularly difficult task. That said, opening up backend interfaces in the frontend, something we've avoided doing, opens other cans of worms and we'd be doing something new and exceptional here. I'm not saying it's bad or wrong, but one can easily see a world in which people want to edit the project, create a content block, change a text, etc., in the frontend following the same logic—it's easier—that we have in this issue. Why do this and not those things?

I'll also add that we haven't heard that this process is "cumbersome," as Matt puts it, from the press, who works with resources extensively.

**Comment by krosemichael** [comment]
I think that the work flow of adding resources to a teaching text is likely different from doing so in a press. I would imagine that when a press puts a text with resources up in Manifold, the author compiles the resources that are going to accompany a text ahead of time, and then the digital editor prepares and adds them to the backend all at once before switching to the front end to annotate them in the text. When adding resources to a teaching text, I think most teachers will be reading through the text in Manifold and asking themselves where students might need clarification, where there might be an article or video that could provide useful background or context or make a link to previous readings or discussions, etc, then finding the resource, switching to the backend to add it by itself, and then switching back to the front end to annotate it in the place where the process started in the text. Adding resources for teaching is I think for most people more tied to reading through the front end than it is for a press, and the way it's set up now necessitates a lot of toggling back and forth between the front end and the back end to complete the text. 

I'll have to think more about this, but I'd argue resources is different from other places where someone might want to make backend changes in the front end because of the way annotating starts in the front end and is so repetitive as a process (and traditionally spontaneous!). You're kind of decorating the text, after all. 

I like the idea of doing it in a drawer within the reader.

**Comment by zdavis** [comment]
I’m amenable to this argument. I wonder if we could just include the first
screen of resource creation (select type and add required fields) in the
drawer. If the user wanted to flesh it out, she would need to go to the
backend.

**Comment by krosemichael** [comment]
Yes I think that's a great idea

**Comment by mkgold** [comment]
An easier, low-impact way to accomplish this, I think, would be to just add a link on the existing add resource drawer on the front end directly to the place on the backend where one can add a resource to a project. In that sense, I'd say that this issue is related to https://github.com/ManifoldScholar/manifold/issues/1960 , as it involves the difficulty of navigating from the frontend of a project back to its backend

**Comment by zdavis** [comment]
That's doable, of course, but I think Krysia makes a pretty strong case for why this should be doable without the user losing her place in the text.

**Comment by mkgold** [comment]
I agree and think it would be wonderful if you think it's reasonable/feasible.
